module/modmanager/manager.go

```diff
@@ -597,6 +596,8 @@ func (mgr *Manager) attemptRestart(ctx context.Context, mod *module) []resource.
 		return orphanedResourceNames
 	}
 
+	// deregister crashed module's resources, and let checkReady reset m.handles.
+	mod.deregisterResources()
```

Comment:
Fun bug here. In module restart logic, we were only deregistering resources of crashed modules when our attempt to re-add a resource to the restarted module failed or when the restart itself failed. We should also deregister resources before `checkReady` to ensure _all_ resources associated with the crashed module are deregistered before being reregistered from the restarted module. I added `m.handles = nil` in `deregisterResources` to ensure this call and the one in the deferred `!success` block above do not conflict (no double-deregistration).